src/mbgl/text/shaping.cpp

```diff
@@ -195,13 +195,62 @@ std::set<std::size_t> determineLineBreaks(const std::u16string& logicalInput,
     return leastBadBreaks(evaluateBreak(logicalInput.size(), currentX, targetWidth, potentialBreaks, 0, true));
 }
 
+struct AnchorAlignment {
```

Comment:
Since this struct is only used as a temporary variable on the stack, how about just inlining `getAnchorAlignment` where it's used and replacing the `AnchorAlignment` variable with separate `horizontalAlign` and `verticalAlign` variables?